db/revision.go

```diff
@@ -230,7 +230,8 @@ const nonJSONPrefix = byte(1)
 // Looks up the raw JSON data of a revision that's been archived to a separate doc.
 // If the revision isn't found (e.g. has been deleted by compaction) returns 404 error.
 func (db *DatabaseCollection) getOldRevisionJSON(ctx context.Context, docid string, revid string) ([]byte, error) {
-	data, _, err := db.Bucket.GetRaw(oldRevisionKey(docid, revid))
+	// FIXME (bbrks): Shouldn't this doc metadata be namespaced by collection?
+	data, _, err := db.dbCtx.MetadataStore.GetRaw(oldRevisionKey(docid, revid))
```

Comment:
> **Warning**
> FIXME @bbrks 

It should... May need to figure out a generic way of namespacing metadata though.